pkg/fs/fs.go

```diff
@@ -530,9 +530,17 @@ func (fs *FileSystem) RemoveXattr(ctx meta.Context, p string, name string) (err
 }
 
 func (fs *FileSystem) lookup(ctx meta.Context, p string, followLastSymlink bool) (fi *FileStat, err syscall.Errno) {
+	var inode Ino
+	var attr = &Attr{}
+
+	if err = fs.m.Resolve(ctx, p, followLastSymlink, &inode, attr); err != syscall.ENOTSUP {
+		return
```

Comment:
If err is 0, we should return an FileStat.

Reply:
We also need the last `name` to create the FileStat, maybe I can list it as another pointer parameter to `Resolve`.

Reply:
The last name could be parsed from `p`.

Reply:
What if we follow a symlink to a different name? This would happen inside the Lua script and we have no idea if it has followed a symlink to a different path, right?

Reply:
We could stop to resolve if any symlink found.

Reply:
Can't we following symlink in the Lua script?

Reply:
This script is only used for performance, so it will enough if it can handle the common case (without a symlink).

Reply:
Also, this is only used for Java SDK, Windows or S3 gateway, there is usually no symlink in these cases.